Pod/Classes/SKYDatabase.m

```diff
@@ -316,69 +317,70 @@ - (void)sky_saveRecords:(NSArray<SKYRecord *> *)records
         }
 
         SKYModifyRecordsOperation *operation =
-            [[SKYModifyRecordsOperation alloc] initWithRecordsToSave:presavedRecords];
+            [[SKYModifyRecordsOperation alloc] initWithRecords:presavedRecords];
         operation.atomic = atomically;
         if (completion) {
             operation.modifyRecordsCompletionBlock =
-                ^(NSArray *savedRecords, NSError *operationError) {
+                ^(NSArray<SKYRecordResult<SKYRecord *> *> *savedRecords, NSError *operationError) {
                     dispatch_async(dispatch_get_main_queue(), ^{
                         completion(savedRecords, operationError);
                     });
                 };
         }
 
-        if (perRecord) {
-            operation.perRecordCompletionBlock = ^(SKYRecord *record, NSError *error) {
-                dispatch_async(dispatch_get_main_queue(), ^{
-                    perRecord(record, error);
-                });
-            };
-        }
-
         [self executeOperation:operation];
     });
 }
 
+#pragma mark Public methods for records
+
 - (void)saveRecord:(SKYRecord *)record completion:(SKYRecordSaveCompletion)completion
 {
     [self sky_saveRecords:@[ record ]
-        atomically:NO
-        completion:^(NSArray *records, NSError *error) {
-            if (error != nil && error.code != SKYErrorPartialOperationFailure) {
-                if (completion) {
-                    completion(nil, error);
-                }
-            }
-        }
-        perRecordCompletion:^(SKYRecord *record, NSError *error) {
-            if (completion) {
-                completion(record, error);
-            }
-        }];
+               atomically:NO
+               completion:^(NSArray<SKYRecordResult<SKYRecord *> *> *results, NSError *error) {
+                   if (!completion) {
+                       return;
+                   }
+                   if (error) {
+                       completion(nil, error);
+                   } else {
+                       completion(results[0].value, nil);
```

Comment:
We encountered the same problem in JS SDK, after discussion I think we should use non-atomic api and passing the `results[0].error` to the completion like delete api. Since in atomic operation, the error wrapped by `AtomicOperationFailure` error and save single record api should not have concept of atomic and non-atomic.